cmd/avrogo/generate.go

```diff
@@ -29,10 +29,7 @@ const nullType = "avrotypegen.Null"
 // schema.RecordDefinition by looking at their match within given parsed namespace
 func shouldImportAvroTypeGen(namespace *parser.Namespace, definitions []schema.QualifiedName) bool {
```

Comment:
I haven't looked in details but that `==` comparison is definitely wrong!

How about something like this instead, which keeps the binary search but (hopefully!) fixes the logic:

```
// shouldImportAvroTypeGen reports whether avrotypegen is required. It checks that the definitions given are of type
// schema.RecordDefinition by looking at their match within given parsed namespace
func shouldImportAvroTypeGen(namespace *parser.Namespace, definitions []schema.QualifiedName) bool {
	for _, def := range namespace.Definitions {
		searchName := def.AvroName().Name
		_, found := sort.Find(len(definitions), func(i int) int {
			return strings.Compare(searchName, definitions[i].Name)
		})
		if found {
			if _, ok := def.(*schema.RecordDefinition); ok {
				return true
			}
			if _, ok := def.(*schema.FixedDefinition); ok {
				return true
			}
		}
	}
	return false
}
```

Note: I haven't actually run the above code :)

Reply:
It worked perfectly! I'll adjust the PR.

Thanks for the help, I really don't have much experience with the `sort` package and I was able to learn some cool stuff! :bowing_man: